src/main/xsl/common/functions.xsl

```diff
@@ -0,0 +1,137 @@
+
+  <xsl:function name="dita-ot:get-element-id" as="xs:string?">
+    <xsl:param name="href"/>
+    <xsl:variable name="fragment" select="substring-after($href, '#')" as="xs:string"/>
+    <xsl:if test="contains($fragment, '/')">
+      <xsl:value-of select="substring-after($fragment, '/')"/>
+    </xsl:if>
+  </xsl:function>
+
+  <xsl:function name="dita-ot:has-element-id" as="xs:boolean">
+    <xsl:param name="href"/>
+    <xsl:sequence select="contains(substring-after($href, '#'), '/')"/>
+  </xsl:function>
+
+  <xsl:function name="dita-ot:get-topic-id" as="xs:string?">
+    <xsl:param name="href"/>
+    <xsl:variable name="fragment" select="substring-after($href, '#')" as="xs:string"/>
+    <xsl:if test="string-length($fragment) gt 0">
+      <xsl:choose>
+        <xsl:when test="contains($fragment, '/')">
+          <xsl:value-of select="substring-before($fragment, '/')"/>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:value-of select="$fragment"/>
+        </xsl:otherwise>
+      </xsl:choose>
+    </xsl:if>
+  </xsl:function>
+
+  <!-- language -->
+
+  <xsl:function name="dita-ot:get-current-language" as="xs:string">
+    <xsl:param name="ctx" as="node()"/>
+
+    <xsl:sequence select="
+      lower-case(($ctx/ancestor-or-self::*[@xml:lang][1]/@xml:lang, $DEFAULTLANG)[1])
+    "/>
+  </xsl:function>
+
+  <xsl:function name="dita-ot:get-iso-language-code" as="xs:string">
+    <xsl:param name="lang" as="xs:string"/>
+    <xsl:sequence select="tokenize($lang, '-')[1]"/>
+  </xsl:function>
+
+  <xsl:function name="dita-ot:get-language-codes" as="xs:string*">
+    <xsl:param name="lang" as="xs:string"/>
+    <xsl:sequence select="$lang, dita-ot:get-iso-language-code($lang)"/>
+  </xsl:function>
+
+  <xsl:function name="dita-ot:get-first-topic-language" as="xs:string">
+    <xsl:param name="ctx" as="node()"/>
+
+    <xsl:sequence select="
+      lower-case(($ctx/*/@xml:lang, $ctx/dita/*/@xml:lang, $DEFAULTLANG)[1])
+    "/>
+  </xsl:function>
+
+  <!-- URI -->
+
+  <xsl:function name="dita-ot:normalize-uri" as="xs:string">
+    <xsl:param name="uri" as="xs:string"/>
+    <xsl:call-template name="dita-ot:normalize-uri">
+      <xsl:with-param name="src" select="tokenize($uri, '/')"/>
+    </xsl:call-template>
+  </xsl:function>
+
+  <xsl:function name="dita-ot:get-variable">
+    <xsl:param name="ctx" as="node()"/>
+    <xsl:param name="id" as="xs:string"/>
+    <xsl:param name="params" as="node()*"/>
+
+    <xsl:call-template name="findString">
+      <xsl:with-param name="ctx" select="$ctx" tunnel="yes"/>
+      <xsl:with-param name="id" select="$id"/>
+      <xsl:with-param name="params" select="$params"/>
+      <xsl:with-param name="ancestorlang"
+        select="dita-ot:get-language-codes(dita-ot:get-current-language($ctx))"/>
+      <xsl:with-param name="defaultlang" select="dita-ot:get-language-codes($DEFAULTLANG)"/>
+    </xsl:call-template>
+  </xsl:function>
+
+  <xsl:function name="dita-ot:get-variable">
```

Comment:
Yeah, I didn't add one for these two because I wasn't 100% sure what it should optimally be. `xs:string*`, I guess? Or should it be `node()*`? `xs:string` would throw a fatal error if DITA-OT can't find the variable, so I don't think we want that.